pkg/ddevapp/drupal.go

```diff
@@ -182,79 +210,153 @@ $databases['default']['default'] = array(
+			return fmt.Errorf("failed to include %s in %s: %v", drupalConfig.SiteSettingsLocal, drupalConfig.SiteSettings, err)
+		}
 	}
 
-	return settingsFilePath, nil
+	return nil
 }
 
-// createDrupal8SettingsFile creates the app's settings.php or equivalent,
-// adding things like database host, name, and password
-// Returns the fullpath to settings file and err
-func createDrupal8SettingsFile(app *DdevApp) (string, error) {
+// writeDrupalCommonSettingsFile creates the app's settings.php or equivalent,
+// which does nothing more than imports the ddev-managed settings.ddev.php.
+func writeDrupalCommonSettingsFile(drupalConfig *DrupalSettings, filePath string) error {
+	tmpl, err := template.New("settings").Funcs(getTemplateFuncMap()).Parse(drupalCommonSettingsTemplate)
+	if err != nil {
+		return err
+	}
+
+	// Ensure target directory is writable.
+	dir := filepath.Dir(filePath)
+	if err = os.Chmod(dir, 0755); err != nil {
+		return err
+	}
 
-	settingsFilePath, err := app.DetermineSettingsPathLocation()
+	// Create file
+	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE, 0644)
 	if err != nil {
-		return "", fmt.Errorf("Failed to get Drupal 8 settings file path: %v", err.Error())
+		return err
+	}
+	defer util.CheckClose(file)
+
+	if err := tmpl.Execute(file, drupalConfig); err != nil {
+		return err
 	}
-	output.UserOut.Printf("Generating %s file for database connection.", filepath.Base(settingsFilePath))
 
+	return nil
+}
+
+// createDrupal7SettingsFile manages creation and modification of settings.php and settings.ddev.php.
+// If a settings.php file already exists, it will be modified to ensure that it includes
+// settings.ddev.php, which contains ddev-specific configuration.
+func createDrupal7SettingsFile(app *DdevApp) (string, error) {
 	// Currently there isn't any customization done for the drupal config, but
 	// we may want to do some kind of customization in the future.
 	drupalConfig := NewDrupalSettings()
 
-	err = writeDrupal8SettingsFile(drupalConfig, settingsFilePath)
-	if err != nil {
-		return settingsFilePath, fmt.Errorf("Failed to write Drupal settings file: %v", err.Error())
+	if err := manageDrupalCommonSettingsFile(app, drupalConfig); err != nil {
+		return "", err
+	}
+
+	// Create settings.ddev.php if one doesn't already exist
+	if fileutil.FileExists(app.SiteLocalSettingsPath) {
+		output.UserOut.Printf("ddev settings file %s already exists", drupalConfig.SiteSettingsLocal)
+	} else {
+		output.UserOut.Printf("No %s exists, creating one", drupalConfig.SiteSettingsLocal)
+		if err := writeDrupal7SettingsFile(drupalConfig, app.SiteLocalSettingsPath); err != nil {
+			return "", fmt.Errorf("failed to write Drupal settings file %s: %v", app.SiteLocalSettingsPath, err)
+		}
+	}
+
+	if err := createGitIgnore(filepath.Dir(app.SiteLocalSettingsPath), drupalConfig.SiteSettingsLocal); err != nil {
+		output.UserOut.Warnf("Failed to write .gitignore: %v", err)
```

Comment:
I don't think this will happen, right? We've already done the permission setting on sites/default before this? Or maybe not, this is at config time and we do that at start time. But anyway, please improve the message to say where the gitignore creation failed (full path of failed creation) (And of course this pattern is repeated elsewhere.)